src/api/searchDevice/src/v1/steps.py

```diff
@@ -1,15 +1,80 @@
+from typing import List
+
 from aws_lambda_powertools.utilities.data_classes import APIGatewayProxyEvent
+from domain.core.device import Device, DeviceType
+from domain.fhir_translation.device import create_fhir_searchset_bundle
+from domain.repository.device_repository import DeviceRepository
+from domain.repository.marshall import unmarshall_value
 from event.step_chain import StepChain
 
-from ..data.response import devices, endpoints
-
 
-def get_results(data, cache):
+# Think we require a validation decorator here.
+def parse_event_query(data, cache):
     event = APIGatewayProxyEvent(data[StepChain.INIT])
-    device_type = event.query_string_parameters["device_type"]
-    return {"product": devices, "endpoint": endpoints}.get(device_type.lower(), {})
+    return (
+        {
+            "query_string": event.query_string_parameters["device_type"].upper(),
+            "host": event.multi_value_headers["Host"],
+        }
+        if event.query_string_parameters["device_type"]
+        else None
+    )
+
+
+def set_device_type(data, cache):
+    event_data = data[parse_event_query]
+    device_query_param = event_data.get("query_string")
+    return (
+        DeviceType.PRODUCT if device_query_param == "PRODUCT" else DeviceType.ENDPOINT
```

Comment:
Would it be better to write this so that its not one or the other - we are very likely to have different devicetypes in the future - could do a map of string to devicetype and get the right devicetype from that?

Reply:
It's not the case now. I guess until that happens we should leave it. That's agile.

Reply:
In theory it won't get this far as there is now validation on the device_type filter. Anything other than "product" or endpoint will get rejected. However I've now set it up to be a map. which can be added to.